redaxo/src/core/lib/packages/package.php

```diff
@@ -366,22 +366,22 @@ public function clearCache()
         }
 
         $cache = rex_file::getCache($path = rex_path::coreCache(self::PROPERTIES_CACHE_FILE));
-        if (!$cache) {
-            return;
-        }
-
-        unset($cache[$this->getPackageId()]);
-
-        if ($this instanceof rex_addon) {
-            $start = $this->getPackageId().'/';
-            foreach ($cache as $packageId => $_) {
-                if (str_starts_with((string) $packageId, $start)) {
-                    unset($cache[$packageId]);
+        if ($cache) {
+            unset($cache[$this->getPackageId()]);
+
+            if ($this instanceof rex_addon) {
+                $start = $this->getPackageId().'/';
+                foreach ($cache as $packageId => $_) {
+                    if (str_starts_with((string) $packageId, $start)) {
+                        unset($cache[$packageId]);
+                    }
                 }
             }
+
+            rex_file::putCache($path, $cache);
         }
 
-        rex_file::putCache($path, $cache);
+        rex_extension::registerPoint(new rex_extension_point('PACKAGE_CACHE_DELETED', $this, [], true));
```

Comment:
Die aktuelle Variante bedeutet, dass man sich zunächst allgemein auf den EP registrieren muss, und dann erst in der Extension das Package prüfen kann:

```php

        rex_extension::register('PACKAGE_CACHE_DELETED', function (rex_extension_point $ep) {
            if ('yrewrite' !== $ep->getSubject()->getPackageId()) {
                return;
            }
            
            // hier der code
        });
```

Ich denke das ist aber ok. Bzw. ich habe nicht wirklich eine bessere Idee.
Dynamische EP-Namen, also Package-ID im Namen, würde ich eher vermeiden, denke ich.

Reply:
Finde ich gut so. Man könnte überlegen eine named EP klasse zu verwenden, dann ist der subject-type ggf auch allen klarer

Reply:
Ja stimmt. Habe ich nun umgesetzt. Beispiel-Code oben in der PR-Beschreibung.
(phpstorm versteht das mit den Templates ja glücklicherweise inzwischen auch. Also phpstorm kennt auch den subject-type.)